src/trixi_timeit.jl

```diff
@@ -0,0 +1,43 @@
+# Enable debug timings `@trixi_timeit timer() "name" stuff...`.
+# This allows us to disable timings completely by executing
+# `TimerOutputs.disable_debug_timings(TrixiBase)`
+# and to enable them again by executing
+# `TimerOutputs.enable_debug_timings(TrixiBase)`
+timeit_debug_enabled() = true
+disable_debug_timings() = TimerOutputs.disable_debug_timings(@__MODULE__)
+enable_debug_timings() = TimerOutputs.enable_debug_timings(@__MODULE__)
+
+# Store main timer for global timing of functions
+const main_timer = TimerOutput()
+
+# Always call timer() to hide implementation details
+timer() = main_timer
+
+#     @trixi_timeit timer() "some label" expression
+#
+# Basically the same as a special case of `@timeit_debug` from
+# [TimerOutputs.jl](https://github.com/KristofferC/TimerOutputs.jl),
+# but without `try ... finally ... end` block. Thus, it's not exception-safe,
+# but it also avoids some related performance problems. Since we do not use
+# exception handling in Trixi, that's not really an issue.
+#
+# Copied from [Trixi.jl](https://github.com/trixi-framework/Trixi.jl).
+macro trixi_timeit(timer_output, label, expr)
```

Comment:
If we're moving it here, we should turn it into a proper API with docstrings.

```suggestion
"""
    @trixi_timeit timer() "some label" expression

Basically the same as a special case of `@timeit_debug` from
[TimerOutputs.jl](https://github.com/KristofferC/TimerOutputs.jl),
but without `try ... finally ... end` block. Thus, it's not exception-safe,
but it also avoids some related performance problems. Since we do not use
exception handling in Trixi, that's not really an issue.
"""
macro trixi_timeit(timer_output, label, expr)
```

It would also be nice to get docstrings for `enable_debug_timings()` and `disable_debug_timings()`, linking all three of them